fvm/state/accounts.go

```diff
@@ -41,6 +42,34 @@ func NewAccounts(stateHolder *StateHolder) *Accounts {
 	}
 }
 
+func (a *Accounts) AllocateStorageIndex(address flow.Address) (uint64, error) {
+	var index uint64
+	indexBytes, err := a.getValue(address, false, KeyStorageIndex)
+	if err != nil {
+		return 0, err
+	}
+
+	if len(indexBytes) == 0 {
+		// if not exist for the first time set it to 1 and return
+		// note that zero is reserved for other puporses. (e.g. empty storageIndex checks)
+		index = 1
+	} else if len(indexBytes) != uint64StorageSize {
+		// this should be fatal
+		return 0, fmt.Errorf("invalid storage index byte size (%d != 8)", len(indexBytes))
+	} else {
+		index = binary.BigEndian.Uint64(indexBytes)
+	}
+
+	newIndexBytes := make([]byte, uint64StorageSize)
+	binary.BigEndian.PutUint64(newIndexBytes, uint64(index+1))
```

Comment:
I'm not familiar with `a.getValue`.  If `indexBytes` is created on the fly by `a.getValue`, maybe `indexBytes` can be reused and replace `newIndexBytes`.

Reply:
yeah I was not sure if reusing indexBytes has any unwanted side effects like values cache, so I didn't risk it,  but I'll add it to the list of optimizations we could do it the future.